packages/cmr-client/CMR.js

```diff
@@ -0,0 +1,301 @@
+    if (params.ummgVersion) headers.Accept = 'application/json';
+
+    return headers;
+  }
+
+  /**
+   * Adds a collection record to the CMR
+   *
+   * @param {string} xml - the collection XML document
+   * @returns {Promise.<Object>} the CMR response
+   */
+  async ingestCollection(xml) {
+    const headers = this.getHeaders({ token: await this.getToken() });
+    return ingestConcept('collection', xml, 'Collection.DataSetId', this.provider, headers);
+  }
+
+  /**
+   * Adds a granule record to the CMR
+   *
+   * @param {string} xml - the granule XML document
+   * @returns {Promise.<Object>} the CMR response
+   */
+  async ingestGranule(xml) {
+    const headers = this.getHeaders({ token: await this.getToken() });
+    return ingestConcept('granule', xml, 'Granule.GranuleUR', this.provider, headers);
+  }
+
+  /**
+   * Adds/Updates UMMG json metadata in the CMR
+   *
+   * @param {Object} ummgMetadata - UMMG metadata object
+   * @returns {Promise<Object>} to the CMR response object.
+   */
+  async ingestUMMGranule(ummgMetadata) {
+    const headers = this.getHeaders({
+      token: await this.getToken(),
+      ummgVersion: ummVersion(ummgMetadata)
+    });
+
+    const granuleId = ummgMetadata.GranuleUR || 'no GranuleId found on input metadata';
+    logDetails.granuleId = granuleId;
+
+    let response;
+    try {
+      await validateUMMG(ummgMetadata, granuleId, this.provider);
+
+      response = await got.put(
+        `${getUrl('ingest', this.provider)}granules/${granuleId}`,
+        {
+          json: true,
+          body: ummgMetadata,
+          headers
+        }
+      );
+      if (response.body.errors) {
+        throw new Error(`Failed to ingest, CMR Errors: ${response.errors}`);
+      }
+    }
+    catch (error) {
+      log.error(error, logDetails);
+      throw error;
+    }
+
+    return response.body;
+  }
+
+  /**
+   * Deletes a collection record from the CMR
+   *
+   * @param {string} datasetID - the collection unique id
+   * @returns {Promise.<Object>} the CMR response
+   */
+  async deleteCollection(datasetID) {
+    const headers = this.getHeaders({ token: await this.getToken() });
+    return deleteConcept('collection', datasetID, headers);
+  }
+
+  /**
+   * Deletes a granule record from the CMR
+   *
+   * @param {string} granuleUR - the granule unique id
+   * @returns {Promise.<Object>} the CMR response
+   */
+  async deleteGranule(granuleUR) {
+    const headers = this.getHeaders({ token: await this.getToken() });
+    return deleteConcept('granules', granuleUR, this.provider, headers);
+  }
+
+  /**
+   * Search in collections
+   *
+   * @param {string} searchParams - the search parameters
+   * @param {string} searchParams.provider_short_name - provider shortname
+   * @param {string} [format=json] - format of the response
+   * @returns {Promise.<Object>} the CMR response
+   */
+  async searchCollections(searchParams, format = 'json') {
```

Comment:
I'm confused.  If we have
`const cmrClient = new CMR({ provider: 'my-provider', clientId: 'my-clientId', username: 'my-username', password: 'my-password' });`
And then `cmrClient.searchCollection or searchGranule`, it is supposed to return the collections or granules only for the `my-provider`, but I don't see that by reading the code.

Reply:
I think this is still needed:
https://github.com/nasa/cumulus/blob/master/packages/cmrjs/cmr.js#L328
const params = Object.assign({}, { provider_short_name: this.provider }, searchParams);

Reply:
I think it's necessary to have some integration tests using the new cmr-client library, to verify it works.

Reply:
@jennyhliu The existing CMR integration tests (which are using the `cmrjs` package) are using the `cmr-client` package, since `cmrjs` calls `cmr-client`.  Are those tests sufficient?

Reply:
But the way the `cmrjs` calls `cmr-client` is different from calling `cmr-client` directly, and  it's not transparent.
https://github.com/nasa/cumulus/pull/842/files#diff-edfb462e041ef85a0cacf11443f5e20dR183

Reply:
This is a change we made in the new package. Instead of taking the provider short name from this.provider, we're taking it as a parameter. 
`@param {string} searchParams.provider_short_name - provider shortname`

Reply:
I feel the user doesn't have to provider searchParams or searchParams.provider_short_name, it should use this.provider for searching.

Reply:
hmmm I don't have a strong preference either way. @yjpa7145 How strongly do you feel about not using `this.provider` ?